src/crypto/sha256.cpp

```diff
@@ -511,6 +520,12 @@ std::string SHA256AutoDetect()
         ret = "sse4";
 #endif
     }
+#elif (defined(__linux__)) && defined(HAVE_POWER8)
+    if (getauxval(AT_HWCAP2) & 0x02000000) {
```

Comment:
Can we move this check to a function in the power8-specific file?

Reply:
Its right below the x86-specific support check, and its a short file, it seems like overkill.